mathoxide-lib/src/views.rs

```diff
@@ -0,0 +1,60 @@
+pub trait ArrayView {
+    fn translate<ListType: AsRef<[usize]>>(&self, idx: ListType) -> usize;
+    fn offset(&self) -> usize;
+    // fn stride(&self) -> &[usize];
+    fn ndims(&self) -> usize;
+    fn size(&self) -> usize;
+    fn shape(&self) -> &[usize];
+}
+
+pub struct SimpleView {
+    _shape: Vec<usize>,
```

Comment:
underscore prefixing in rust is generally used to denote deliberately unused variables. So just naming the field `shape` should be ok. Fields are private by default in rust.